Compilation failure on MinGW-w64 (x86_64)

`building 'borg.hashindex' extension`
`C:\msys64\mingw64\bin/x86_64-w64-mingw32-gcc.exe -DNDEBUG -march=x86-64 -mtune=generic -O2 -pipe -D_FORTIFY_SOURCE=2 -IC:/msys64/mingw64/include/python3.5m -c borg/hashindex.c -o build/temp.mingw-3.5/borg/hashindex.o`
`In file included from borg/hashindex.c:433:0:`
`borg/_hashindex.c:24:2: error: #error Unknown byte order`
` #error Unknown byte order`
 ` ^~~~~`
`borg/_hashindex.c: In function 'hashindex_index':`
`borg/_hashindex.c:108:12: warning: implicit declaration of function '_le32toh' [-Wimplicit-function-declaration]`
`     return _le32toh(*((uint32_t *)key)) % index->num_buckets;`
`            ^~~~~~~~`
`borg/_hashindex.c: In function 'hashindex_lookup':`
`borg/_hashindex.c:76:15: warning: implicit declaration of function '_htole32' [-Wimplicit-function-declaration]`
` #define EMPTY _htole32(0xffffffff)`
`               ^`
`borg/_hashindex.c:84:100: note: in expansion of macro 'EMPTY'`
` #define BUCKET_IS_EMPTY(index, idx) (*((uint32_t *)(BUCKET_ADDR(index, idx) + index->key_size)) == EMPTY)`
`                                                                                                    ^~~~~`
`borg/_hashindex.c:118:12: note: in expansion of macro 'BUCKET_IS_EMPTY'`
 `        if(BUCKET_IS_EMPTY(index, idx))`
`            ^~~~~~~~~~~~~~~`
`error: command 'C:\\msys64\\mingw64\\bin/x86_64-w64-mingw32-gcc.exe' failed with exit status 1`
`==> ERROR: A failure occurred in build().`
 `   Aborting...`

--- a/borg/_hashindex.c
+++ b/borg/_hashindex.c
@@ -13,10 +13,12 @@
 #endif
 
 #if (defined(BYTE_ORDER)&&(BYTE_ORDER == BIG_ENDIAN)) ||  \
+    (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)) || \
     (defined(_BIG_ENDIAN)&&defined(__SVR4)&&defined(__sun))
 #define _le32toh(x) __builtin_bswap32(x)
 #define _htole32(x) __builtin_bswap32(x)
 #elif (defined(BYTE_ORDER)&&(BYTE_ORDER == LITTLE_ENDIAN)) || \
+      (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) || \
       (defined(_LITTLE_ENDIAN)&&defined(__SVR4)&&defined(__sun))
 #define _le32toh(x) (x)
 #define _htole32(x) (x)

--- a/borg/_hashindex.c
+++ b/borg/_hashindex.c
@@ -12,18 +12,24 @@
 #include <sys/isa_defs.h>
 #endif
 
-#if (defined(BYTE_ORDER)&&(BYTE_ORDER == BIG_ENDIAN)) ||  \
+#if (defined(BYTE_ORDER) && (BYTE_ORDER == BIG_ENDIAN)) ||  \
     (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)) || \
-    (defined(_BIG_ENDIAN)&&defined(__SVR4)&&defined(__sun))
-#define _le32toh(x) __builtin_bswap32(x)
-#define _htole32(x) __builtin_bswap32(x)
-#elif (defined(BYTE_ORDER)&&(BYTE_ORDER == LITTLE_ENDIAN)) || \
+    (defined(_BIG_ENDIAN) && defined(__SVR4)&&defined(__sun))
+#define BORG_BIG_ENDIAN 1
+#elif (defined(BYTE_ORDER) && (BYTE_ORDER == LITTLE_ENDIAN)) || \
       (defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)) || \
-      (defined(_LITTLE_ENDIAN)&&defined(__SVR4)&&defined(__sun))
-#define _le32toh(x) (x)
-#define _htole32(x) (x)
+      (defined(_LITTLE_ENDIAN) && defined(__SVR4)&&defined(__sun))
+#define BORG_BIG_ENDIAN 0
 #else
 #error Unknown byte order
+#endif
+
+#if BORG_BIG_ENDIAN
+#define _le32toh(x) __builtin_bswap32(x)
+#define _htole32(x) __builtin_bswap32(x)
+#else
+#define _le32toh(x) (x)
+#define _htole32(x) (x)
 #endif
 
 #define MAGIC "BORG_IDX"

